libraries/chain/block_header_state.cpp

```diff
@@ -79,6 +77,30 @@ const finalizer_policy& block_header_state::get_last_proposed_finalizer_policy()
    return *active_finalizer_policy;
 }
 
+// The last pending finalizer policy if none pending then the active finalizer policy
+// Used to populate last_pending_finalizer_policy_digest which is expected to be the highest generation pending
+const finalizer_policy& block_header_state::get_last_pending_finalizer_policy() const {
+   if (!finalizer_policies.empty()) {
+      [[maybe_unused]] auto highest_pending_generation = [this]() {
```

Comment:
You can add a comment indicating thus lambda is used only when DEBUG is turned on.